src/server/middleware/repoproxy/proxy.go

```diff
@@ -106,14 +106,25 @@ func handleManifest(w http.ResponseWriter, r *http.Request, next http.Handler) e
 	if err != nil {
 		return err
 	}
-	useLocal, err := proxyCtl.UseLocalManifest(ctx, art, remote)
+	useLocal, man, err := proxyCtl.UseLocalManifest(ctx, art, remote)
+
 	if err != nil {
 		return err
 	}
 	if useLocal {
+		if man != nil {
+			w.Header().Set("Content-Type", man.ContentType)
+			w.Header().Set("Docker-Content-Digest", man.Digest)
+			w.Header().Set("Etag", man.Digest)
+			w.Write(man.Content)
```

Comment:
You will return the content even if it's a HEAD request?

Reply:
Updated